docs/user-guide/configuring.md

```diff
@@ -344,7 +344,7 @@ The second way to use configuration files is via `.eslintrc` and `package.json`
 
 In each case, the settings in the configuration file override default settings.
 
-## Configuration Cascading and Hierarchy
+## Configuration Cascading
```

Comment:
Because the section was becoming very long, and it seemed to make more sense to break it apart into two sections (added section for Hierarchy here: https://github.com/eslint/eslint/pull/2924/files#diff-f92aa5edd27b97e8ba555fe7b8806310R403).  Can re-combine if you prefer.